addons/grenadethrowing/functions/fnc_pfh.sqf

```diff
@@ -0,0 +1,179 @@
+
+    // These magazine checks are potentially not needed, but maaaaaybe would become relevant in some situations, like someone taking a grenade from your pack if you were holding it.
+    if (GVAR(LastGrenadeTypeChecked) == "") then {
+        GVAR(ActiveGrenadeType) = _currentGrenade;
+
+        if (_currentGrenadeMagazineType in (magazines player)) then {
+            GVAR(ActiveGrenadeItem) = _currentGrenade createVehicleLocal ((vehicle player) modelToWorldVisual [0,.3,1.6]);
+            if (GVAR(ActiveGrenadeType) == "") exitWith {
+                GVAR(CancelThrow) = false;
+                GVAR(GrenadeInHand) = false;
+            };
+            GVAR(ActiveGrenadeItem) enableSimulation false;
+        } else {
+            _exit = true;
+        };
+    };
+
+    if (_exit) exitWith {
+        ["Trying to create a grenade we do not have in inventory (1)"] call FUNC(exitThrowMode);
+    };
+
+    if (GVAR(LastGrenadeTypeChecked) != _currentGrenade) then {
+        deleteVehicle GVAR(ActiveGrenadeItem);
+        GVAR(ActiveGrenadeType) = _currentGrenade;
+        if (_currentGrenadeMagazineType in (magazines player)) then {
+            GVAR(ActiveGrenadeItem) = _currentGrenade createVehicleLocal ((vehicle player) modelToWorldVisual [0,.3,1.6]);
+            if (GVAR(ActiveGrenadeType) == "") exitWith {
+                GVAR(CancelThrow) = false;
+                GVAR(GrenadeInHand) = false
+            };
+            GVAR(CurrentThrowSpeed) = getNumber (configFile >> "CfgMagazines" >> _currentThrowable select 0 >> "initSpeed");
+            GVAR(ActiveGrenadeItem) enableSimulation false;
+        } else {
+            _exit = true
+        };
+    };
+
+    if (_exit) exitWith {
+        ["Trying to create a grenade we do not have in inventory (2)"] call FUNC(exitThrowMode);
+    };
+
+    GVAR(CameraOffset) = [0, 0, 0.3];
+    private _posLeftOffset = 0;
+    private _posRightOffset = 0;
+
+    if (inputAction "leanLeft" > 0.1) then {
+        _posLeftOffset = linearConversion [0, 1, inputAction "leanLeft", 0, 0.25, true];
+        GVAR(CameraOffset) = [-_posLeftOffset, 0, 0.3];
+    };
+    if (inputAction "leanRight" > 0.1) then {
+        _posRightOffset = linearConversion [0, 1, inputAction "leanRight", 0, 0.25, true];
+        GVAR(CameraOffset) = [_posRightOffset, 0, 0.3];
+    };
+
+    private _xAdjust_Bonus = 0;
+    private _yAdjust_Bonus = 0;
+    if (GVAR(ThrowType) == "under") then {
+        _xAdjust_Bonus = -0.075;
+        _yAdjust_Bonus = 0.1;
+    };
+
+    GVAR(CameraOffset) set [0, (GVAR(CameraOffset) select 0) + GVAR(xAdjust) + _xAdjust_Bonus];
+    GVAR(CameraOffset) set [1, (GVAR(CameraOffset) select 1) + GVAR(yAdjust) + _yAdjust_Bonus];
+    GVAR(CameraOffset) set [2, (GVAR(CameraOffset) select 2) + GVAR(zAdjust)];
+
+    private _posFin = positionCameraToWorld GVAR(CameraOffset);
+    if (!(surfaceIsWater _posFin)) then {
```

Comment:
The ASLtoAGL handles things correctly over water too, so this if should be removed.